pkg/ceph/rbd/rbd.go

```diff
@@ -29,16 +29,15 @@ const (
 )
 
 func storeKey(key string) (string, error) {
+	err := os.MkdirAll(tmpKeyFileLocation, 0700)
+	if err != nil {
+		return "", fmt.Errorf("error creating a temporary directory: %w", err)
+	}
+
 	tmpfile, err := ioutil.TempFile(tmpKeyFileLocation, tmpKeyFileNamePrefix)
 	if err != nil {
 		return "", fmt.Errorf("error creating a temporary keyfile: %w", err)
 	}
-	defer func() {
-		if err != nil {
-			// don't complain about unhandled error
-			_ = os.Remove(tmpfile.Name())
-		}
-	}()
```

Comment:
this is still valid incase of error

Reply:
in the above error, we are returning so not sure if that part we'll reach.

Reply:
yes both someone needs to clean up the file right? who is doing that one

Reply:
@subhamkrai that defer is to clean up the keyfile in case anything else goes wrong in the function

Reply:
got it.